lib/instrumentation/instrumentor.js

```diff
@@ -0,0 +1,77 @@
+'use strict';
+
+var utils = require('../utils');
+
+var defaultOnConfig = {
+  console: true
+};
+
+var defaultConfig = {
+  console: false,
+  http: false,
+  pg: false
+};
+
+function fill(obj, name, replacement, track) {
+  var orig = obj[name];
+  obj[name] = replacement(orig);
+  if (track) {
+    track.push([obj, name, orig]);
+  }
+}
+
+function instrument(Raven, config) {
+  if (config === false) {
+    return;
+  } else if (config === true) {
+    config = defaultOnConfig;
+  } else {
+    config = utils.extend({}, defaultConfig, config);
+  }
+
+  Raven.instrumentedOriginals = [];
+  Raven.instrumentedModules = [];
+
+  var Module = require('module');
+  fill(Module, '_load', function (origLoad) {
+    return function (moduleId, parent, isMain) {
+      var origModule = origLoad.apply(this, arguments);
```

Comment:
Maybe, but there are a few other spots where we're doing `something.apply(this, arguments)` in instrumentation for console and http (and fwiw raven-js also has spots where it doesn't do the copy opt...), so I'd rather go all-or-nothing on it.